File: CppPractice/CppLeetcode/Leetcode2.cpp
#include<vector>


struct ListNode
{
    int val;
    ListNode *next;
    ListNode(): val(0),next(nullptr){};
    ListNode(int x): val(x) ,next(nullptr){};
    ListNode(int x, ListNode *p): val(x), next(p){};
};

class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) 
    {
        int carry=0;
        ListNode *root= new ListNode();
        ListNode *cur = root;
        ListNode *cur1=l1;
        ListNode *cur2=l2;
        while(cur1||cur2||carry)
        {
            int digit1=0;
            int digit2=0;
            if(cur1)
            {
                digit1 = cur1->val;
                cur1 = cur1->next;
            }
            if(cur2)
            {
                digit2 = cur2->val;
                cur2 = cur2->next;
            }
            cur->next = new ListNode((digit1+digit2+carry)%10);
            cur =cur->next;
            carry = (digit1+digit2+carry)/10;

        }
        return root->next;
    }
};

File: Pythonleetcode/Leetcode2.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        def get_num(root):
            if not root:
                return 0
            return root.val+10*get_num(root.next)

        sum_result=get_num(l1)+get_num(l2)
        str_sum_result=str(sum_result)
        len_result=len(str(sum_result))
        add_index=len_result-1
        ans_link = ListNode(0)
        p=ans_link
        while add_index>=0:
            p.next=ListNode(int(str_sum_result[add_index]))
            add_index -=1
            p=p.next

        return ans_link.next


